ethers-core/src/types/trace/geth/pre_state.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    types::{Address, H256, U256},
+    utils::from_int_or_hex_opt,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+// https://github.com/ethereum/go-ethereum/blob/91cb6f863a965481e51d5d9c0e5ccd54796fd967/eth/tracers/native/prestate.go#L38
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum PreStateFrame {
+    Default(PreStateMode),
+    Diff(DiffMode),
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PreStateMode(pub BTreeMap<Address, AccountState>);
```

Comment:
I think this should probably b `#[serde(transparent)]`?

https://serde.rs/container-attrs.html#transparent

Reply:
Seems transparent is enabled for newtype by [default](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=226ffebdc45dc392ae006722284b62da)?

Reply:
TIL